doc/dev/PROTOCOL.md

```diff
@@ -485,6 +485,25 @@ The result is returned as a list of:
 }
 ```
 
+### `refactor-extract-region -start <position> -stop <position> -extract-name <name>`
+
+```
+    -start <position>  Where extracted region start
+    -stop <position>  Where extracted region end
+    -extract-name <name>  Name used for the generated let binding
+```
+
+Returns the string `Nothing to do` (if extractor is not ables to select an expression to extract in the given position interval) or the following object:
```

Comment:
Returning a string seems to be an issue, we should take look at other commands in order to throw a proper exception>

Reply:
Nice! Thanks!